swap_meet/vendor.py

```diff
@@ -1,2 +1,69 @@
+from .item import Item
+
+
 class Vendor:
-    pass
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_category(self, category):
+        result = []
+        for item in self.inventory:
+            if item.category == category:
+                result.append(item)
+        return result
+
+    def swap_items(self, vendor_friend, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor_friend.inventory:
+            return False
+        self.remove(my_item)
+        vendor_friend.add(my_item)
+        vendor_friend.remove(their_item)
+        self.add(their_item)
+        return True
+
+    def swap_first_item(self, vendor_friend):
+        if self.inventory == [] or vendor_friend.inventory == []:
+            return False
+        vendor_friend.add(self.inventory[0])
+        self.remove(self.inventory[0])
+        self.add(vendor_friend.inventory[0])
+        vendor_friend.remove(vendor_friend.inventory[0])
+        return True
+
+    def get_best_by_category(self, category):
+        inventory_in_category = []
+        for item in self.inventory:
+            if item.category == category:
+                inventory_in_category.append(item)
+        if len(inventory_in_category) == 0:
+            return None
```

Comment:
you could dry this code up by calling the instance method `get_by_category` here instead of rewriting the code out!